src/Adapter/MSTest.CoreAdapter/TestMethodFilter.cs

```diff
@@ -95,5 +95,34 @@ internal object PropertyValueProvider(TestCase currentTest, string propertyName)
 
             return null;
         }
+
+        /// <summary>
+        /// Gets filter expression from run context.
+        /// </summary>
+        /// <param name="context">Run context</param>
+        /// <returns>Filter expression.</returns>
+        private ITestCaseFilterExpression GetTestCaseFilterFromRunContext(IRunContext context)
+        {
+            return context.GetTestCaseFilter(this.supportedProperties.Keys, this.PropertyProvider);
+        }
+
+        /// <summary>
+        /// Gets filter expression from discovery context.
+        /// </summary>
+        /// <param name="context">Discovery context</param>
+        /// <returns>Filter expression.</returns>
+        private ITestCaseFilterExpression GetTestCaseFilterFromDiscoveryContext(IDiscoveryContext context)
+        {
+            try
+            {
+                // GetTestCaseFilter is present in DiscoveryContext but not in IDiscoveryContext interface.
+                MethodInfo methodGetTestCaseFilter = context.GetType().GetRuntimeMethod("GetTestCaseFilter", new[] { typeof(IEnumerable<string>), typeof(Func<string, TestProperty>) });
+                return (ITestCaseFilterExpression)methodGetTestCaseFilter?.Invoke(context, new object[] { this.supportedProperties.Keys, (Func<string, TestProperty>)this.PropertyProvider });
+            }
+            catch (TargetInvocationException ex)
+            {
+                throw ex.InnerException;
```

Comment:
and who would handle this if it isn't a TestPlatformException or for that matter a non-TargetInvocationException is thrown?

Reply:
platform handles it automatically. message is logged in case of exception by platform.